### PR TITLE
Centralizing accept/content-type headers management

### PR DESCRIPTION
**Pull request by massa84**

### CODE REVIEW

**Comment by customcommander** [comment]
I think that the two "tabs to spaces" could be squashed together (_interactive rebase all the things!_ :smile_cat: ). Otherwise this PR makes a lot of sense to me :+1:

File: lib/headers.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+define(function() {
+
+    'use strict';
+
+    /*
```
Comment (paulwib) [inline lib/headers.js]: This example of extracting the content-types could go in a separate utility script.

File: lib/headers.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    });
+    console.log('var contentTypes = ' + JSON.stringify(contentTypes).replace(/"/g,'\'').replace(/,/g,',\n    ') + ';');
+
+
+
+    and pasting the result in this file
+
+    */
+
+    var contentTypes = {
```
Comment (paulwib) [inline lib/headers.js]: Please edit this list to include only the public content-types used by the SDK.

**Comment by paulwib** [comment]
@massa84 please can you add unit tests of the `Headers` methods.

**Comment by paulwib** [comment]
@customcommander that is kind of different as it gets the headers based on the file type rather than being a specific API content-type.

**Comment by paulwib** [comment]
I'm going to close this one as incomplete and code has drifted somewhat. If anyone feels it's still something valuable feel free to re-open a new PR.

**Pull request closed without merging**